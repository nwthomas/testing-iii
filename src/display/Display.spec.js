import React from "react";
import { render } from "react-testing-library";
import renderer from "react-test-renderer";
import "react-testing-library/cleanup-after-each"; // Prevents memory leaks after each test
import "jest-dom/extend-expect";

import Display from "./Display";

describe("<Display />", () => {
  it("renders the Display component to the screen without crashing", () => {
    const helpers = render(<Display closed={false} locked={false} />);
  });

  it("matches the snapshot of Display", () => {
    const tree = renderer.create(<Display />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render as open and unlocked onto the screen when passed appropriate props", () => {
    const { getByText } = render(<Display closed={false} locked={false} />);
    const unlockedDisplay = getByText(/Unlocked/i);
    const openDisplay = getByText(/Open/i);
    expect(unlockedDisplay).toBeDefined();
    expect(openDisplay).toBeDefined();
  });

  it("should render as closed and locked onto the screen when passed appropriate props", () => {
    const { getByText } = render(<Display closed={true} locked={true} />);
    const unlockedDisplay = getByText(/Locked/i);
    const openDisplay = getByText(/Closed/i);
    expect(unlockedDisplay).toBeDefined();
    expect(openDisplay).toBeDefined();
  });

  it("changes classes for different colors when the closed/locked props' boolean value changes", () => {
    const { getByText, rerender } = render(
      <Display closed={false} locked={false} />
    );
    const unlockedDisplay = getByText(/Unlocked/i);
    const openDisplay = getByText(/Open/i);
    expect(unlockedDisplay.className).toBe("led green-led");
    expect(openDisplay.className).toBe("led green-led");
    rerender(<Display closed={true} locked={true} />);
    const lockedDisplay = getByText(/Locked/i);
    const closedDisplay = getByText(/Closed/i);
    expect(lockedDisplay.className).toBe("led red-led");
    expect(closedDisplay.className).toBe("led red-led");
  });
});
