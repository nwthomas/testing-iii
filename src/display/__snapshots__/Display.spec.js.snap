// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches the snapshot of Display 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
